pandas/core/sorting.py

```diff
@@ -285,6 +319,93 @@ def nargsort(
     return indexer
 
 
+def apply_key(index, key, level=None):
```

Comment:
type these

Reply:
So this is tricky. I can't import MultiIndex into `sorting.py` because it creates a circular dependency that makes everything fail. I couldn't think of a way around this.